### PR TITLE
Feat/Address fields on the vendor registration form

### PR DESCRIPTION
**Pull request by tanjiralmamun**
### All Submissions:

* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

### Changes proposed in this Pull Request:

Added the Store Address fields on the vendor registration form with an admin setting.

Closes #.

### How to test the changes in this Pull Request:

1. Enable the `Enable Address Fields` option from the `Dokan > General` setting screen
2. Registration as a vendor from the `My Account` page to see the changes

Screenshots:
image
image

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [ ] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [ ] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

### CODE REVIEW

**Comment by shashwatahalder01** [comment]
Scenario:
location input fields should follow woocommerce selling location settings

Steps to reproduce:
1. restrict selling location to specific countries
2. go to vendor reg from

Screenshot:
af0
af1

**Comment by shashwatahalder01** [comment]
Scenario:
profile progress bar is not increasing if completed form reg form

Steps to reproduce:
1. complete address from reg from
2. see progress completion bar

Screenshot:
af3

**Comment by shashwatahalder01** [comment]
Scenario:
address input fields aren't on vendor reg shortcode

Screenshot:
af4

**Comment by shashwatahalder01** [comment]
Scenario:
If the address field is enabled on the vendor reg form, users get two address input forms: one on the reg form and another on the vendor setup wizard which is redundant. There should be only one address input field for vendor sign-up.

Screenshot:
af2
af5

File: assets/js/vendor-address.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+( function($){
+    var dokan_address_wrapper = $( '.dokan-address-fields' );
```
Comment (nurul-umbhiya) [inline assets/js/vendor-address.js:2]: use let/const instead of var while possible,

File: templates/global/seller-registration-form.php [hunk 2]
```diff
@@ -30,9 +30,19 @@
...
+    </p>                                   
 
     <?php
+
+    /**
+     * Store Address Fields
+     */
+    $is_address_enabled = dokan_get_option( 'enabled_address_on_reg', 'dokan_general', 'on' );
+
+    if ( $is_address_enabled == 'on' ) {
```
Comment (nurul-umbhiya) [inline templates/global/seller-registration-form.php:42]: directly use `if ( 'on' === dokan_get_option( 'enabled_address_on_reg', 'dokan_general', 'on' ); ) `

File: templates/account/vendor-registration.php [hunk 3]
```diff
@@ -56,6 +56,16 @@
     </p>
 
     <?php
+
+    /**
+     * Store Address Fields
+     */
+    $is_address_enabled = dokan_get_option( 'enabled_address_on_reg', 'dokan_general', 'on' );
+
+    if ( $is_address_enabled == 'on' ) {
```
Comment (nurul-umbhiya) [inline templates/account/vendor-registration.php:65]: directly use `if ( 'on' === dokan_get_option( 'enabled_address_on_reg', 'dokan_general', 'on' ) )`

File: includes/Assets.php [hunk 4]
```diff
@@ -539,6 +539,11 @@ public function get_scripts() {
                 'deps'    => [ 'jquery', 'dokan-vue-vendor' ],
                 'version' => filemtime( $asset_path . 'js/product-category-ui.js' ),
             ],
+            'dokan-vendor-address'      => [
```
Comment (nurul-umbhiya) [inline includes/Assets.php:542]: fix all phpcs related issues

File: templates/global/seller-registration-form.php (hunk 2 above)
Comment (nurul-umbhiya) [reply templates/global/seller-registration-form.php:42]: Also default value should be `off` as you set `off` as default value on corresponding settings

File: assets/js/vendor-address.js (hunk 1 above)
Comment (nurul-umbhiya) [reply assets/js/vendor-address.js:2]: Have you removed the corresponding duplicate js code from codebase?

**Review by nurul-umbhiya — CHANGES_REQUESTED** [review]

**Review by nurul-umbhiya — APPROVED** [review]

**Pull request merged**